server/src/main/java/io/druid/segment/realtime/plumber/CommitterMaker.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.segment.realtime.plumber;
+
+import io.druid.data.input.Committer;
+
+/**
+ * Object that can be used to make a Committer.
+ * <p/>
+ * This is useful when something *may* want to make a Committer, but also may not, and when making the actual
+ * Committer could be expensive.
+ */
+public interface CommitterMaker
```

Comment:
If there is no real reason, then I would possibly be inclined to not have another interface and use Supplier<Committer>